Task 5

enginire4ek opened the pull request:
- Added the function of viewing pins
- The function of adding pins is in progress

enginire4ek committed:
Task 5
- Added the function of viewing pins
- The function of adding pins is in progress

enginire4ek committed:
Task 5. Updates
- Linter fixes
- Small fixes in design and improving the quality of the code

102Rudy commented on data/src/main/java/com/example/pinterbest/data/repository/PinsRepositoryImpl.kt:
```diff
@@ -43,10 +51,34 @@ class PinsRepositoryImpl
+                    pinImage.toRequestBody("image/*".toMediaTypeOrNull(), 0, pinImage.size)
+                )
+
+                val pinResponse = ApiClient().getInstance().getClient()
```
it would be better to provide `client` via constructor as other dependencies

102Rudy commented on domain/src/main/java/com/example/pinterbest/domain/entities/PinInfo.kt:
```diff
@@ -0,0 +1,8 @@
+    val title: String,
+    val description: String,
+    val tags: List<String>?,
+    val boardID: Int = 246
```
what is 246? Please, avoid magic numbers

enginire4ek committed:
Task 5. Updates
- Added new repository: BoardRepository
- Added the ability to create boards
- Added the ability to select a board when creating a pin
- Added list of pins from "Saved Pins" board in profile-screen
